src/signed.rs

```diff
@@ -206,15 +201,15 @@ pub struct IncompleteMultisignatureError<'a, T: Signable, MK: MultiKeychain> {
 }
 
 impl<'a, T: Signable, MK: MultiKeychain> PartiallyMultisigned<'a, T, MK> {
-    pub fn sign(signable: T, keychain: &'a MK) -> Self {
+    pub fn sign(signable: Indexed<T>, keychain: &'a MK) -> Self {
```

Comment:
This shouldn't require an `Indexed` argument, anything can be signed and the wrapper is not even used for anything.

Reply:
You can, you should just pass the underlying type here, not the wrapped type you had to pass to create the `Signed` object.

I actually had a discussion about this with Michał and we ended up creating a low priority task to make explicit usage of `Indexed` completely unnecessary.

Reply:
`add_signature` takes `&mut PartiallyMultisigned<..., T, ...>` and `Signed<..., Indexed<T>, ...>`. Your version is a bit redundant, and you probably should anyway assert that `signable.index` is equal to `keychain.index()`

Reply:
Meh...you're absolutely right. I totally misunderstood some compiler error and followed it wrongly. I've reverted the change.